Source/Core/DolphinNoGUI/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+if(NOT(USE_X11 OR ENABLE_HEADLESS))
```

Comment:
And macOS. I remember trying to build nogui on those and it failed since it didn't have any code to create a window or render context (except for some X11 code). I wonder if we could use a 3rd party lib to simplify that maybe?